### PR TITLE
Revise Write Xml

### PR DESCRIPTION
**Pull request by jhorbulyk**
Resolves https://github.com/elasticio/xml-component/issues/30

### CODE REVIEW

File: component.json [hunk 1]
```diff
@@ -13,12 +13,65 @@
...
       "metadata": {
         "in": {},
-        "out": "./lib/schemas/jsonToXml.out.json"
+        "out": "./lib/schemas/jsonToXmlOld.out.json"
+      }
+    },
+    "jsonToXmlV2": {
+      "title": "JSON to XML",
+      "main": "./lib/actions/jsonToXml.js",
+      "description": "Takes the result of a JSONata expression and creates corresponding XML as either a string or an attachment",
```
Comment (kirill-levitskiy) [inline component.json:28]: There is the new help links section: https://docs.elastic.io/guides/configuring-help-links.html
`description` text is not displaying on the frontend anymore.

File: lib/actions/jsonToXml.js [hunk 2]
```diff
@@ -1,69 +1,69 @@
...
-  return eioUtils.newMessageWithBody({
-    xmlString: result,
-  });
-}
+  if (!uploadToAttachment) {
+    this.logger.info('Sending XML data in message.');
+    await this.emit('data', messages.newMessageWithBody({
+      xmlString,
+    }));
+    return;
```
Comment (kirill-levitskiy) [inline lib/actions/jsonToXml.js:43]: `await` can be replaced by `return` 
```
return this.emit('data', messages.newMessageWithBody({
      xmlString,
    }));
```

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "xml",
   "version": "1.2.1",
```
Comment (kirill-levitskiy) [inline package.json:3]: You forgot to bump minor version of the component. This update should be described in short form in the `CHANGELOG.md` as well.

File: component.json [hunk 4]
```diff
@@ -13,12 +16,71 @@
...
+      },
+      "metadata": {
+        "in": {
+          "type": "object",
+          "properties": {
+            "input": {
+              "title": "JSON to convert",
+              "type": "object",
+              "required": true,
+              "maxLength": 1000
```
Comment (kirill-levitskiy) [inline component.json:60]: Is it enough 1000 symbols as max length of JSON input messages?

Comment (jhorbulyk) [reply component.json:60]: This is taking advantage of some undocumented platform behavior.  It is currently being used by the email component: https://github.com/elasticio/email-component/blob/master/component.json#L52 When there is a large enough `maxLength` value, the mapper will cause the JSONata input to be rendered on more than one line.  Thoughts on using this functionality?

File: README.md [hunk 5]
```diff
@@ -1,19 +1,45 @@
...
+The inner-text of an XML element can also be controlled with `#` sub-object.
+
+For example:
+```json
+{
+  "someTag": {
+    "_attr": {
+      "id": "my id"
+    },
+    "_": "my inner text"
```
Comment (A3a3e1) [inline README.md:31]: So '_' or '#'?

Comment (jhorbulyk) [reply README.md:31]: [As per the documentation `charKey` defaults to `_` though it was `#` in earlier versions.](https://www.npmjs.com/package/xml2js#options) We can change it to whatever we want though...

File: component.json (hunk 4 above)
Comment (kirill-levitskiy) [reply component.json:60]: I didn't know about this behavior.
This limitation doesn't allow to put text longer than 1000 symbols in the Integrator mode.
All other use cases is still working, like JSONata mode and mapping from previous steps.

Pros: Looks better, allows to use multi-line editor.
Cons: Integrator mode doesn't allow a text longer than 1000 symbols.

File: component.json [hunk 6]
```diff
@@ -13,12 +16,70 @@
       }
     },
     "jsonToXml": {
-      "title": "JSON to XML",
-      "main": "./lib/actions/jsonToXml.js",
+      "title": "JSON to XML (Deprecated)",
```
Comment (stas-fomenko) [inline component.json:19]: maybe it makes sense remove `(Deprecated)` from the title, as we already have `"deprecated": true`?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,11 @@
+## 1.3.0 (April 8, 2020)
```
Comment (stas-fomenko) [inline CHANGELOG.md:1]: as now we have one release notes per platform and components maybe it makes sense to date changes same as platform monthly release (May 7)? or at least we need to update date as sprint last day (April 23)

File: component.json [hunk 8]
```diff
@@ -1,6 +1,9 @@
 {
   "title": "XML",
-  "description": "Component to work with XML files",
```
Comment (stas-fomenko) [inline component.json:3]: is there any specific reason why the component description is removed?
image

File: component.json [hunk 9]
```diff
@@ -13,12 +16,70 @@
...
         "in": {},
-        "out": "./lib/schemas/jsonToXml.out.json"
+        "out": "./lib/schemas/jsonToXmlOld.out.json"
+      }
+    },
+    "jsonToXmlV2": {
+      "title": "JSON to XML",
+      "main": "./lib/actions/jsonToXml.js",
+      "help": {
+        "link":  "#json-to-xml",
```
Comment (stas-fomenko) [inline component.json:32]: after docs on docs.elastic.io updated would need to check if this link still valid

File: component.json (hunk 8 above)
Comment (jhorbulyk) [reply component.json:3]: I'll change that back. I guess that component level descriptions still appear even though action/trigger level descriptions are not longer rendered. Asking for follow up: https://elasticio.slack.com/archives/C6G1E4WB1/p1587374522293600

Comment (jhorbulyk) [reply component.json:3]: Platform issue: https://github.com/elasticio/elasticio/issues/3816

**Pull request merged**